class/label.rb

```diff
@@ -0,0 +1,27 @@
+class Label
+  attr_reader :id, :title, :color, :items
+
+  def initialize(title, color, items = [], id = Random.rand(100_000))
+    @id = id
+    @title = title
+    @color = color
+    @items = items
+  end
+
+  def add_item(item)
+    @items << item
+    item.label = self
+  end
+end
+
+class Book
+  attr_accessor :id, :publish_date, :publisher, :cover_state, :label
+
+  def initialize(publish_date, publisher, cover_state)
+    @id = Random.rand(100_000)
+    @publish_date = publish_date
+    @publisher = publisher
+    @cover_state = cover_state
+    @label = nil
+  end
+end
```

Comment:
Hello @Dorcas126,

You have done a beautiful job implementing this class. However, I noticed your use of association (labels) and attribute accessors.

**Attribute Accessors:** Use attr_accessor only when you need both read and write access to an attribute. If an attribute is read-only, as in the case of id, use attr_reader instead.

**Association:** Ensure that the association between labels and books is working as intended in the context of our program. Make sure that adding a book to a label's items list also updates the book's label attribute. You should also improve your code by using `SecureRandom` to auto generate ids, and also create a new label method that monitors changes.
an implementation of this could look like:

```suggestion
require 'securerandom'

class Label
  attr_reader :id, :title, :color, :items

  def initialize(title, color, items = [], id = SecureRandom.uuid)
    @id = id
    @title = title
    @color = color
    @items = items
  end

  def add_item(item)
    @items << item
    item.label = self
  end
end

class Book
  attr_accessor :id, :publish_date, :publisher, :cover_state
  attr_reader :label

  def initialize(publish_date, publisher, cover_state, label = nil)
    @id = SecureRandom.uuid
    @publish_date = publish_date
    @publisher = publisher
    @cover_state = cover_state
    @label = label
  end

  def label=(new_label)
    @label&.items&.delete(self)
    @label = new_label
    new_label&.items << self
  end
end
```